File: src/main/java/com/tppartdeux/edoctorat/dto/professeur/ExaminerResponse.java
package com.tppartdeux.edoctorat.dto.professeur;

import com.tppartdeux.edoctorat.dto.candidat.CandidatResponse;
import lombok.AllArgsConstructor;
import lombok.Builder;
import lombok.Data;
import lombok.NoArgsConstructor;

/**
 * Examiner response DTO matching Angular's Examiner interface
 */
@Data
@NoArgsConstructor
@AllArgsConstructor
@Builder
public class ExaminerResponse {
    private Long id;
    private SujetResponse sujet;
    private String cne;
    private Float noteDossier;
    private Integer noteEntretien;
    private String decision;
    private Long commission;
    private CandidatResponse candidat;
    private Boolean publier;
}

File: front-end/src/models/ExaminerResponse.ts
import { Sujet } from './Sujet';
import { Candidat } from './Candidat';

export interface ExaminerResponse {
  id: number;
  sujet: Sujet;
  cne: string;
  noteDossier: number;
  noteEntretien: number;
  decision: string;
  commission: number;
  candidat: Candidat;
  publier: boolean;
}
